### ISSUE TITLE
sheep missing texture

### ISSUE THREAD

**Issue by cigumo** [cigumo/krli#1086]
![](https://cdn.discordapp.com/attachments/567811170342535188/670272391082934272/20200124232233_1.jpg)

**Comment by sunnydaler** [comment]
https://streamable.com/m4148
https://streamable.com/ovet6
[sheep missing texture.txt](https://github.com/cigumo/krli/files/4112767/sheep.missing.texture.txt)

**Comment by cigumo** [comment]
go_enemies_common is loaded and yet it's not found in the atlas db


```
[82135.7946] image_db.ERROR    s() - Sprite sheep_0001 was not found in the atlas db.
Loaded images ------------------
achievements-1    1024
encyclopedia-20    260
go_decals-1    1024
go_decals-2    1024
go_enemies_common-1    2048
go_enemies_wastelands-1    1024
go_enemies_wastelands-2    1024
go_hero_oni-1    1024
go_stage10_bg-1    1920
go_towers-1    2048
go_towers-2    2048
gui_common-1    2048
gui_common-2    2048
gui_notifications_bg-1    900
gui_notifications_common-1    1024
gui_notifications_common-2    1024
gui_portraits-1    1024
gui_portraits-2    1024
view_options-1    1024

Atlas usage---------------------
go_stage10_bg-0.711111:1
view_options-0.711111:1
gui_notifications_bg-0.711111:1
encyclopedia-0.711111:1
gui_notifications_common-0.711111:1
go_enemies_common-0.711111:6
gui_common-0.711111:1
gui_portraits-0.711111:1
go_stage10-0.711111:1
go_towers-0.711111:6
go_hero_oni-0.711111:1
achievements-0.711111:1
go_stages_wastelands-0.711111:1
go_decals-0.711111:6
go_enemies_wastelands-0.711111:1

Counts---------------------
Total images: 19 (133.60205078125 MB)
Total deferred images: 97
Total frames: 7678

love.graphics.getStats()---
self:   table: 0x0c3188c0
    ['canvases']        ->  13
    ['canvasswitches']  ->  0
    ['drawcalls']       ->  0
    ['fonts']           ->  36
    ['images']          ->  20
    ['shaderswitches']  ->  0
    ['texturememory']   ->  148266664


```

**Comment by sunnydaler** [comment]
works in 4.0.04

**Comment by cigumo** [comment]
The error happens after removing and adding a frame with the same key in image_db.db_atlas.

- In kr1-desktop-4.0.04 I tested that by having no textures with the same name the error does not happen
- In kr1-desktop-4.0.03b I tested doing collectgarbage twice after removing items from db_atlas, and it worked. 

The error seems related to these two commits: 
- https://github.com/LuaJIT/LuaJIT/commit/046129dbdda5261c1b17469a2895a113d14c070a
- https://github.com/LuaJIT/LuaJIT/commit/de48d000941f58a8d4d816e9d227de8a4f3c9de6

**Comment by cigumo** [comment]
Found the problem. Not properly unloading the required_textures in game.lua, and then the map removes the sheep frames. 

Fixed in 4.0.05
